File: WebUI/src/app/models/feed.ts
import { FeedDetailComponent } from './../feed/feed-detail/feed-detail.component';
import { FeedTypeEnum } from './feedTypeEnum';
export class Feed {
    public id: string;
    public userName: string;
    public heading: string;
    public data: string;
    public imageUrl: string;
    public type: FeedTypeEnum;
    public followups: Feed;
    public dateCreated: Date;
    public dateLastUpdated: Date;

    constructor(id: string,
         userName: string,
         heading: string,
         data: string,
         imageUrl: string,
         type: FeedTypeEnum,
         followups: Feed,
         dateCreated: Date,
         dateLastUpdated: Date) {
             this.id = id;
             this.userName = userName;
             this.heading = heading;
             this.data = data;
             this.imageUrl = imageUrl;
             this.type = type;
             this.followups = followups;
             this.dateCreated = dateCreated;
             this.dateLastUpdated = dateLastUpdated;
    }
}

File: backend/Services/MainService/Models/Feed.cs
using System;
using System.Runtime.Serialization;
using core.repository.azureCosmos;
using Newtonsoft.Json;

namespace MainService.Models
{
    public class Feed : DocumentBase //: MainService.Models.IFeed
    {
        [JsonProperty("userName")]
        public string UserName { get; set; }
        [JsonProperty("heading")]
        public string Heading { get; set; }
        [JsonProperty("data")]
        public string Data { get; set; }
        [JsonProperty("imageUrl")]
        public string ImageUrl { get; set; }
        [JsonProperty("type")]
        public FeedTypes Type { get; set; }
        [JsonProperty("followups")]
        public Feed[] Followups { get; set; }
        [JsonProperty("dateCreated")]
        public DateTimeOffset DateCreated { get; set; }
        [JsonProperty("dateLastUpdated")]
        public DateTimeOffset DateLastUpdated { get; set; }

        public override string ToString()
        {
            return JsonConvert.SerializeObject(this);
        }
    }
}